sklearn/utils/seq_dataset.pxd.tp

```diff
@@ -0,0 +1,83 @@
+{{py:
+
+"""
+    Template file for easily generate fused types consistent code using Tempita
+    (https://github.com/cython/cython/blob/master/Cython/Tempita/_tempita.py).
+
+    Generated file: seq_dataset.pxd
+
+    Each class is duplicated for all dtypes (float and double). The keywords
+    between double braces are substituted in setup.py.
+"""
+
+# name, c_type
+dtypes = [('', 'double'),
```

Comment:
I agree. But doing that we would need to recode sag_fast.pyx for example... we could put it off to another PR (#9020) or just rename ``SequentialDataset`` in sag_fast.pyx as ``SequentialDataset64``.